base_domain_operator/tests/test_base_domain_operator.py

```diff
@@ -32,3 +37,14 @@ def test_base_domain_operator(self):
             ('name', 'substring_of', ' YourCompany is great')
         ])
         self.assertTrue(substring_matches)
+        # see how _operator_parent_of deals with strings
+        leaf = self.env['base.domain.operator']._operator_parent_of(
```

Comment:
any reason to do complicated stuff here instead of just constructing a domain as above?